functions/src/error-handling.ts

```diff
@@ -0,0 +1,55 @@
+import { randomUUID } from "crypto";
+import { Response } from "firebase-functions";
+import { ResponseType } from "./utils";
+
+const SHARE_LINK_ERRORS = {
+  400: "Invalid Request.",
+  404: "Not Found.",
+  500: "Internal API Failure.",
+};
+type ShareLinkErrorCode = keyof typeof SHARE_LINK_ERRORS;
+
+/**
+ * Custom error handling class for share link API errors to log errors internally
+ * and send a user-friendly error message to the client.
+ *
+ * @param code HTTP status code for the error.
+ * @param response Firebase functions response to send external error to.
+ * @param external Error message to send to the client. Message should not include error code.
+ * @param internal Error or error message to log to the console. If not provided, externalMessage is used.
+ * @param responseType Response type to send to the client. Defaults to text/html.
+ */
+export class ShareLinkError extends Error {
+  constructor(
+    errorCode: ShareLinkErrorCode,
+    response: Response<unknown>,
+    external: string,
+    internal?: string | Error,
+    responseType: ResponseType = ResponseType.TEXT
```

Comment:
I'm not entirely sure I'm following why we have two different response types and how you'd decide which one to use...

Reply:
Yeah it was a little contrived--I've gone ahead and removed them.

Some of the endpoints (`/registerUrl` and `/shareLinksByUrl`) return JSON, while others return html/text, so the idea was to match the error response type to the matching endpoint response type. But, this seems unnecessary and adds a layer of confusion, and it seems returning text/html for all errors is okay